modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/io/experimental/Metadata.scala

```diff
@@ -241,7 +241,9 @@ object Metadata {
     )
 
   def recalculate(previous: EventsToEntities, events: List[EnrichedEvent]): EventsToEntities =
-    previous ++ events.map(e => MetadataEvent(e) -> unwrapEntities(e))
+    (previous.toList ++ events.map(e => MetadataEvent(e) -> unwrapEntities(e)))
+      .groupBy(_._1)
+      .map { case (k, v) => k -> v.flatMap(_._2).toSet }
```

Comment:
How about using [cats semigroup syntax](https://typelevel.org/cats/typeclasses/semigroup.html) to do the hard work of merging the map of sets:

```
previous |+| events.map(e => MetadataEvent(e) -> unwrapEntities(e)).toMap
```

Reply:
Nice but the `.toMap` is also a problem because it doesn't concatenate values

Reply:
I could create a list of `Map` of size 1 though